src/group/service.ts

```diff
@@ -0,0 +1,69 @@
+import database from '../database';
+import { User } from '../user/types';
+
+type GroupRecord = {
```

Comment:
I'm thinking there's still a lot of duplication here compared to the Users service. Maybe it's easier to just move the service functions in there?